lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,154 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Recipe from "./recipe";
+import { RecipeInterface, APIOptions, APIInterface } from "./types";
+import { MultiFactorAuthClaim } from "./multiFactorAuthClaim";
+import { SessionContainerInterface } from "../session/types";
+import Multitenancy from "../multitenancy";
+import { getUser } from "../..";
+import { getUserContext } from "../../utils";
+
+export default class Wrapper {
+    static init = Recipe.init;
+
+    static MultiFactorAuthClaim = MultiFactorAuthClaim;
+
+    static async checkAllowedToSetupFactorElseThrowInvalidClaimError(
+        session: SessionContainerInterface,
+        factorId: string,
+        userContext?: Record<string, any>
+    ) {
+        let ctx = getUserContext(userContext);
+        const user = await getUser(session.getUserId(), ctx);
+        if (!user) {
+            throw new Error("UKNKNOWN_USER_ID");
+        }
+        const factorsSetup = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getFactorsSetupForUser({
+            user,
+            userContext: ctx,
+        });
+        const mfaClaimValue = await session.getClaimValue(MultiFactorAuthClaim, ctx);
+        const completedFactors = mfaClaimValue?.c ?? {};
+        const defaultMFARequirementsForUser = await Recipe.getInstanceOrThrowError().recipeInterfaceImpl.getRequiredSecondaryFactorsForUser(
+            {
+                userId: session.getUserId(),
+                userContext: ctx,
+            }
+        );
+
+        const tenantInfo = await Multitenancy.getTenant(session.getTenantId(), userContext);
+        const defaultMFARequirementsForTenant: string[] = tenantInfo?.requiredSecondaryFactors ?? [];
```

Comment:
if the input tenant doesn't exist, why are we not throwing an error? What happens in other functions in other recipes if the input tenant doesn't exist? Also, in the recipe.ts file version of this function, you check if a tenant doesn't exist, but here you don't. Strange..

If you are throwing an error here, you should not throw an unauthorised error.

Reply:
added throw